src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -784,7 +784,7 @@ private void transitiveCovers(Set<Symbol> covered) {
             }
         }
 
-        private boolean isTransitivelyCovered(Symbol sealed, Set<Symbol> covered) {
+        private boolean isTransitivelyCovered(Type seltype, Symbol sealed, Set<Symbol> covered) {
             DeferredCompletionFailureHandler.Handler prevHandler =
```

Comment:
Why do we need to guard against completion failures here in Flow?

Reply:
Consider a case like this:
```
sealed interface I permits A, B {}
//A and B defined
...
I i = null;
switch (i) {
     case A -> {}
}
```

This will produce an error, because the switch is not exhaustive. Now, consider, in addition, that `B` is missing. It feels a bit harsh to produce an error about the missing `B` permitted subclass (possibly producing that error instead of the exhaustiveness error), as it seems plausible permitted subclasses may be missing during compilation.

Reply:
Uhm - I get what you say. But isn't sealing meant to be used within a single maintenance domain? If B is missing, how the heck did we get to Flow in the first place (since for sealed classes we check that permitted subclasses extend the sealed class being defined)

Reply:
> Uhm - I get what you say. But isn't sealing meant to be used within a single maintenance domain? If B is missing, how the heck did we get to Flow in the first place (since for sealed classes we check that permitted subclasses extend the sealed class being defined)

Ok - maybe you are considering the case where `I` is used in a compilation unit which did not defined it. And which cannot (for some reason) see B.  Seems cornery, but what I'm mostly worried about is that it's not (I think) uniform with the rest of the javac code dealing with sealed classes. For instance, look at the cast conversion rules in `Types::areDisjoint` - in that case there's no protection against missing types.

I think the compiler code should be consistent in how we reason about sealed permitted types?

Reply:
Right, the case is where the sealed class/interface is used from some other project. It is somewhat cornery, but might happen even for us - consider an sealed API interface in the JDK, which has permitted classes that are in an implementation package. The `--release` data normally do not include non-API packages, so when looking at the API interface, we may encounter a class that does not exist.

I didn't realize there is the same issue with `areDisjoint`, but you are right it suffers from the problem and that it should be consistent. I'll take a look at that, possibly separately, if that's OK.

Reply:
Yep - fine to postpone!